indexer/test/advertisement-walker.test.js

```diff
@@ -517,3 +523,249 @@ describe('data schema for REST API', () => {
+      /No entries found/
+    )
+  })
+
+  it('throws an error for unsupported codec', () => {
+    // Use a CID with an unsupported codec
+    const unsupportedCid = 'bafkreigrnnl64xuevvkhknbhrcqzbdvvmqnchp7ae2a4ulninsjoc5svoq'
+
+    assert.throws(
+      () => processEntries(unsupportedCid, dagJsonChunk),
+      /Unsupported codec/
+    )
+  })
+
+  // Data integrity test using real multihash operations
+  it('correctly creates a CID from entry data', () => {
+    // Process the entries
+    const result = processEntries(dagJsonCid, dagJsonChunk)
+
+    // Create the expected CID directly
+    const expectedCid = CID.create(1, 0x55, mh).toString()
+
+    // They should match
+    assert.strictEqual(result, expectedCid, 'CID should match the one created directly')
+  })
+
+  // Test with entries from CBOR encoding/decoding
+  it('correctly handles DAG-CBOR entries serialized with @ipld/dag-cbor', () => {
+    // Process the entries
+    const result = processEntries(dagCborCid, dagCborChunk)
+
+    // Create the expected CID directly
+    const expectedCid = CID.create(1, 0x55, mh).toString()
+
+    // They should match
+    assert.strictEqual(result, expectedCid, 'CID should match the one created directly')
+  })
+
+  // Error case tests with real data
+  it('handles malformed base64 in DAG-JSON gracefully', () => {
+    const malformedChunk = {
+      Entries: [
+        {
+          '/': {
+            bytes: 'This-is-not-valid-base64!'
+          }
+        }
+      ]
+    }
+
+    // We expect an error when processing this malformed data
+    assert.throws(
+      () => processEntries(dagJsonCid, malformedChunk),
+      /Incorrect length/
+    )
+  })
+
+  it('handles invalid multihash in DAG-CBOR gracefully', () => {
+    const invalidChunk = {
+      Entries: [new Uint8Array([0, 1, 2, 3])] // Too short to be a valid multihash
+    }
+
+    // We expect an error when processing this invalid multihash
+    assert.throws(
+      () => processEntries(dagCborCid, invalidChunk),
+      /Incorrect length/
+    )
+  })
+})
+
+describe('fetchCid', () => {
+  // Use a real DAG-JSON CID that will naturally have codec 0x0129 (297)
+  // CIDs that start with 'bagu' are DAG-JSON encoded
+  const dagJsonCid = 'baguqeerayzpbdctxk4iyps45uldgibsvy6zro33vpfbehggivhcxcq5suaia'
+  // Sample JSON response
+  const testResponse = { test: 'value' }
+  // Use a real DAG-CBOR CID that will naturally have codec 0x71 (113)
+  // CIDs that start with 'bafy' are DAG-CBOR encoded
+  const dagCborCid = 'bafyreictdikh363qfxsmjp63i6kup6aukjqfpd4r6wbhbiz2ctuji4bofm'
+
+  it('uses DAG-JSON codec (0x0129) to parse response as JSON', async () => {
+    // Mock fetch to return JSON
+    // @ts-ignore
+    const mockFetch = mock.fn(() => Promise.resolve({
+      ok: true,
+      status: 200,
+      json: () => Promise.resolve(testResponse),
+      arrayBuffer: () => { throw new Error('Should not call arrayBuffer for JSON') }
+    }))
```

Comment:
I consider `@ts-ignore` almost always a code smell indicating we are taking a shortcut. Think of unsafe code in Rust. In other languages, it's a common practice to write a comment justifying the shortcut - explain what is the TS error and why it's safe to ignore it.

I'd like us to do the same when we use `@ts-ignore`. 

It will also make it easier to review pull requests because I am struggling to see what errors you are trying to suppress without any additional comments.

Now, setting ts-ignore aside.

Do we still need to wrap the custom fetch function in `mock.fn()`? What's the benefit of such approach when compared to the following:

```suggestion
    /** @type {typeof fetch} */
    const mockFetch = () => Promise.resolve({
      ok: true,
      status: 200,
      json: () => Promise.resolve(testResponse),
      arrayBuffer: () => { throw new Error('Should not call arrayBuffer for JSON') }
    })
```

The same comment applies to all other places below where `@ts-ignore` or `mock.fn` is used.